vizro-core/examples/scratch_dev/app.py

```diff
@@ -1,49 +1,247 @@
 
+import time
+import yaml
+
+import dash
 import pandas as pd
+from flask_caching import Cache
+
 import vizro.models as vm
 import vizro.plotly.express as px
 from vizro import Vizro
-from vizro._themes._color_values import COLORS
-
-pastry = pd.DataFrame(
-    {
-        "pastry": [
-            "Scones",
-            "Bagels",
-            "Muffins",
-            "Cakes",
-            "Donuts",
-            "Cookies",
-            "Croissants",
-            "Eclairs",
-            "Brownies",
-            "Tarts",
-            "Macarons",
-            "Pies",
-        ],
-        "Profit Ratio": [-0.10, -0.15, -0.05, 0.10, 0.05, 0.20, 0.15, -0.08, 0.08, -0.12, 0.02, -0.07],
-    }
+from vizro.managers import data_manager
+from functools import partial
+
+print("INITIALIZING")
+
+SPECIES_COLORS = {"setosa": "#00b4ff", "versicolor": "#ff9222", "virginica": "#3949ab"}
+BAR_CHART_CONF = dict(x="species", color="species", color_discrete_map=SPECIES_COLORS)
+SCATTER_CHART_CONF = dict(x="sepal_length", y="petal_length", color="species", color_discrete_map=SPECIES_COLORS)
+
+# Relevant for the "page_6" only
+FILTER_COLUMN = "species"
+# FILTER_COLUMN = "sepal_length"
+# FILTER_COLUMN = "date_column"
+
+
+def load_from_file(filter_column=None, parametrized_species=None):
+    # Load the full iris dataset
+    df = px.data.iris()
+    df["date_column"] = pd.date_range(start=pd.to_datetime("2024-01-01"), periods=len(df), freq="D")
+
+    if parametrized_species:
+        return df[df["species"].isin(parametrized_species)]
+
+    with open("data.yaml", "r") as file:
+        data = yaml.safe_load(file)
+        data = data or {}
+
+    filter_column = filter_column or FILTER_COLUMN
+    if filter_column == "species":
+        final_df = pd.concat(
+            objs=[
+                df[df[filter_column] == "setosa"].head(data.get("setosa", 0)),
+                df[df[filter_column] == "versicolor"].head(data.get("versicolor", 0)),
+                df[df[filter_column] == "virginica"].head(data.get("virginica", 0)),
+            ],
+            ignore_index=True,
+        )
+    elif filter_column == "sepal_length":
+        final_df = df[df[filter_column].between(data.get("min"), data.get("max"), inclusive="both")]
+    elif filter_column == "date_column":
+        date_min = pd.to_datetime(data.get("date_min"))
+        date_max = pd.to_datetime(data.get("date_max"))
+        final_df = df[df[filter_column].between(date_min, date_max, inclusive="both")]
+    else:
+        raise ValueError("Invalid FILTER_COLUMN")
+
+    return final_df
+
+
+data_manager["load_from_file"] = load_from_file
+data_manager["load_from_file_species"] = partial(load_from_file, filter_column="species")
+data_manager["load_from_file_sepal_length"] = partial(load_from_file, filter_column="sepal_length")
+data_manager["load_from_file_date_column"] = partial(load_from_file, filter_column="date_column")
+
+
+# TODO-DEV: Turn on/off caching to see how it affects the app.
```

Comment:
I was also glad so many times for some flexibility we implemented even though we didn't know why it would be important in the future.

One of the examples is this line:
`_get_unfiltered_data_targets = list(set(figure_targets + control_targets_targets))`  from the (https://github.com/mckinsey/vizro/pull/879#discussion_r1843819953).

It wouldn't work if we decided to return target objects as a `list` of objects (without mapping them in the `dict: [target_id, Any]` targets), when we developed actions system. 😃